### ISSUE TITLE
EPSFStar.register_epsf might be incorrect.

### ISSUE THREAD

**Issue by SunilSimha** [astropy/photutils#805]
I encountered this bug when trying to use `compute_reidual_image` for an `EPSFStar` object. I noticed the default computed residuals were essentially the same. Looking at `register_epsf` (line 235) of epsf_star.py, I think I see the error. The code currently reads:
```python
return (self.flux * x_oversamp * y_oversamp *
                epsf.evaluate(xx, yy, flux=1.0, x_0=0, y_0=0))
```
However, `xx` and `yy` already seem to be multiplied by `x_oversamp` and `y_oversamp` factors earlier (lines 230 and 231) and the residuals come correctly if those multipliers are removed.

**Comment by mattabern** [comment]
I just discovered this same problem. The issue is that the `FittableImageModel.evaluate()` method takes a `use_oversampling` parameter that is defaulted to True to re-scale the x and y coordinates. 

In the `register_epsf()` source, the re-scaling is already done in the lines:
```
    yy, xx = np.indices(self.shape, dtype=np.float)
    xx = x_oversamp * (xx - self.cutout_center[0])
    yy = y_oversamp * (yy - self.cutout_center[1])
```
, so it's effectively done twice. This can be fixed by 

changing the code quoted by SunilSimha to:

```
return (self.flux * x_oversamp * y_oversamp *
                epsf.evaluate(xx, yy, flux=1.0, x_0=0, y_0=0, use_oversampling=False))
```

**Comment by larrybradley** [comment]
Fixed in #885.  Thanks!
